src/instructlab/model/serve.py

```diff
@@ -116,6 +116,13 @@ def serve(
         click.secho(f"Failed to determine backend: {e}", fg="red")
         raise click.exceptions.Exit(1)
 
+    if backend == backends.LLAMA_CPP and ctx.args:
+        # need to check before stderr/stdout are redirected, issue #1738
+        ctx.fail(
```

Comment:
I still think failing is not the right approach here. I think we should just log that llama-cpp does not support extra args. `ctx.args` is not even passed down to the backend code.

Reply:
How about we let that PR fix the `ignore_unknown_options` and let https://github.com/instructlab/instructlab/pull/1719 handle the rest?

Reply:
I'm not a big fan of #1719. It's hard coding assumptions. I would rather delegate error checking to vLLM process or use `vllm.EngineArgs` to check for unsupported arguments.

Reply:
I know the hardcoded values are not ideal but at least they give the user in indication that something is wrong. It's not breaking anything, nor changing any existing behavior.